### PR TITLE
Fix EZP-23969: Refactor: Slots should be shared by all Search Engines

### PR DESCRIPTION
**Pull request by pspanja**
This PR resolves https://jira.ez.no/browse/EZP-23969, sub-task of https://jira.ez.no/browse/EZP-23940

This moves Slots into a common namespace and adapts the configuration, making them shared by all search engines (bar Legacy).

### CODE REVIEW

File: eZ/Publish/API/Repository/Tests/SetupFactory/LegacyElasticsearch.php [hunk 1]
```diff
@@ -59,7 +59,7 @@ protected function getServiceContainer()
             $containerBuilder->addCompilerPass( new Compiler\Search\Elasticsearch\AggregateSortClauseVisitorContentPass() );
             $containerBuilder->addCompilerPass( new Compiler\Search\Elasticsearch\AggregateSortClauseVisitorLocationPass() );
             $containerBuilder->addCompilerPass( new Compiler\Search\Solr\FieldRegistryPass() );
-            $containerBuilder->addCompilerPass( new Compiler\Search\Solr\SignalSlotPass() );
+            $containerBuilder->addCompilerPass( new Compiler\Search\SignalSlotPass() );
```
Comment (andrerom) [inline eZ/Publish/API/Repository/Tests/SetupFactory/LegacyElasticsearch.php]: May I suggest we create a `Common` namespace?
If we do this we redo the mistake done in Persistence having several folders which are not implementation of a engine at root, messes up the convention of <API/SPI-part>/Implementation organization in Core.

File: eZ/Publish/Core/Search/Slot.php [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
- * File containing the Solr\Slot abstract class
+ * This file is part of the eZ Publish Kernel package
  *
  * @copyright Copyright (C) eZ Systems AS. All rights reserved.
  * @license For full copyright and license information view LICENSE file distributed with this source code.
  * @version //autogentag//
  */
 
-namespace eZ\Publish\Core\Search\Solr;
+namespace eZ\Publish\Core\Search;
```
Comment (andrerom) [inline eZ/Publish/Core/Search/Slot.php]: May I suggest we create a `Common` namespace or something?
If we do this we redo the mistake done in Persistence having several folders which are not implementation of a engine at root, messes up the convention of <API/SPI-part>/Implementation organization in Core.

Comment (pspanja) [inline eZ/Publish/Core/Search/Slot.php]: Yes, it makes sense. I'll update.

Comment (pspanja) [inline eZ/Publish/Core/Search/Slot.php]: Moved to `Core\Search\Common` in https://github.com/ezsystems/ezpublish-kernel/commit/3b5c3f3580be8311d951ad22cb34f4036c0426eb.

**Comment by pspanja** [comment]
Now ready for review, ping @bdunogier, @lolautruche, @glye, @yannickroger.

**Comment by andrerom** [comment]
+1

**Comment by lolautruche** [comment]
+1

**Comment by pspanja** [comment]
Thanks!

**Pull request merged**